process/src/process.cpp

```diff
@@ -374,6 +374,30 @@ int create(lua_State* L)
     return lua_yield(L, 0);
 }
 
+int homedir(lua_State* L)
+{
+    char homedir_buff[255];
+    size_t homedir_size = 255;
```

Comment:
I almost wonder if we should consider having an optional argument for the max length? On one hand, it’s frustratingly low-level, and it would also prevent us from e.g. implementing some automatic resizing or something, but on the other hand, the behavior of it just dying with 256 byte paths is pretty bad (if they’re UTF-16, which I think they might be on Windows, that’s only 128 characters).

Reply:
After looking at the docs again, it seems that libuv will return the required length if it's not enough.